tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/OrganisationControllerTests.cs

```diff
@@ -36,17 +60,30 @@ public async Task GetAllAsync_NoOrganisationsExist_EmptyResultIsReturned()
             organisationResult.Organisations.Count().Should().Be(0);
         }
 
-        [TestCase(null, null)]
-        [TestCase("Organisation One", null)]
-        [TestCase(null, "ODS 1")]
-        [TestCase("Organisation One", "ODS 1")]
-        public async Task GetAllAsync_SingleOrganisationExists_ReturnsTheOrganisation(string name, string ods)
+        [TestCase(null, null, null, false, false)]
+        [TestCase("Organisation One", null, null, false, false)]
+        [TestCase(null, "ODS 1", null, false, false)]
+        [TestCase("Organisation One", "ODS 1", null, false, false)]
+        [TestCase("Organisation One", "ODS 1", null, true, false)]
+        [TestCase("Organisation One", "ODS 1", null, true, true)]
+        public async Task GetAllAsync_SingleOrganisationExists_ReturnsTheOrganisation(string name, string ods, string primaryRoleId, bool catalogueAgreementSigned, bool hasAddress)
         {
             var organisationId = Guid.NewGuid();
 
             using var controller = OrganisationControllerBuilder
                 .Create()
-                .WithListOrganisation(new List<Organisation>(){new Organisation(organisationId, name, ods)})
+                .WithListOrganisation(new List<Organisation>()
+                {
+                    new Organisation()
```

Comment:
Just to reiterate what I said on the phone. The OrganisationBuilder class could be used here, so that the code contains one area for creating a new Organisation objects. If the Organisation class changes, hopefully there will only be the builder to alter, rather than multiple tests.